### ISSUE TITLE
Better syncthing integration 

### ISSUE THREAD

**Issue by edmundmiller** [orgzly-revived/orgzly-android-revived#104]
I think syncthing is pretty popular amount the org-mode using crowd. 

Just wondering if the revival could have better integration somehow. 

It works, until it doesn't because of the updating of the notebooks or whatever. I've never used any of the other syncing options so I'm not sure what the experience is like.

**Comment by mortezadadgar** [comment]
Yeah adding a option for disabling all syncing features like check file conflicts altogether would be good to have

**Comment by paulelms** [comment]
I use syncthing with orgzly and now with orgzly-revived, I don't understand what problems you are talking about.

**Comment by mortezadadgar** [comment]
Orgzly is not updating synced files when their content changes so you might not get notifications on your mobile when you added a reminder in your PC
Also removed files stays in orgzly

**Comment by amberin** [comment]
Orgzly has never supported automatic discovery of remote changes; something must tell Orgzly to sync.

I would like to add a "sync at regular intervals" feature, but haven't started work on it.

The handling of remotely deleted files changed in v1.8.24. Any feedback/thoughts on those changes is very welcome.

**Comment by rijkt** [comment]
I don't know how realistic this is, but I would prefer my synced/remote file to be the source of truth for the app. I.e. the app being an editor for a file, which can be changed at will without the need for manually resolving conflicts.

**Comment by ShuoShen** [comment]
Yeah, this is my top pain point now, and I'm looking to fix it. 

@edmundmiller , @mortezadadgar , @amberin , @rijkt , @paulelms I'm curious, do you all use Dropbox, WebDAV, or local file for syncing? The mechanism of "listening" to remote changes could be very different. Knowing the use pattern would help myself prioritize the work.

**Comment by amberin** [comment]
@ShuoShen I use Git repositories for syncing.

This issue does not (yet) document what needs improving when it comes to Syncthing integration.

If we're talking about automatic discovery of remote changes, then add I said, all repo types behave the same: sync is only initiated by Orgzly.

> The mechanism of "listening" to remote changes could be very different. Knowing the use pattern would help myself prioritize the work.

I hope that any such work would focus on the local directory repository type. That is the most essential sync functionality, and also what most people rely on (e.g. Syncthing users).

**Comment by ShuoShen** [comment]
Ah, thanks for the pointer. Synthing looks like a good replacement for my Dropbox solution. Curious though, do we have data metrics to tell how many users are using local repo vs Dropbox? I suppose Dropbox dev console should give us some usage data, but not sure if we're tracking local repo. 

BTW, in the codebase, there's a github repo implementation. Is that dependent on synthing for syncing or is it a different repo setup?

**Comment by amberin** [comment]
> I suppose Dropbox dev console should give us some usage data, but not sure if we're tracking local repo.

Yes, we can see the amount of Dropboxa ccounts linked to our app, but we have no other usage metrics whatsoever.

> BTW, in the codebase, there's a github repo implementation. Is that dependent on synthing for syncing or is it a different repo setup?

It's a separate repository type, still in beta. It's not available the Play store ("premium") build, but can be enabled under "Developer settings" in the F-droid build.

**Comment by ShuoShen** [comment]
@amberin do you mind share a rough percentage of users on Dropbox? I'm interested in knowing how worthy it is to continue investing on Dropbox syncing

**Comment by amberin** [comment]
These are all the analytics we have from Dropbox:

![Image](https://github.com/user-attachments/assets/5892fc35-4ca7-429d-96ef-c200ab7d38c8)

**Comment by rijkt** [comment]
@ShuoShen I use a directory sync repository. The chosen directory is inside of a folder managed by syncthing.

**Comment by amberin** [comment]
(I just realized that our test pipeline uses the same app ID as the actual app, so the spikes in API calls the last few days was probably me running tests... I have now fixed this by creating a separate app identity for tests.)

**Comment by ShuoShen** [comment]
@amberin thanks for sharing. 

Based on the data, Dropbox users make up about 30% (320 out of 1100) of total Play Store users. It’s not the majority, but it’s a decent chunk—still feels worth improving for.

**Comment by pjones** [comment]
@amberin Would you accept a PR that responded to the Syncthing-Fork notification as described above?  I might have some time this summer to put that together.

**Comment by amberin** [comment]
@pjones Absolutely!

**Comment by alensiljak** [comment]
I recently started using Syncthing for Orgzly files because Git was not convenient for quick exchange between two devices.
I prefer the separation of functionality between Orgzly and synchronization software.
Like some other participants, I am also puzzled by the issues mentioned here and find it hard to see how the process could be improved further. Orgzly syncs data with a local folder and leaves it to 3rd-party software to do as they please. Sync back to pick up any changes in the local files. My workflow is the same when I'm using git (with PuppyGit or lazygit, for example).
The sync in Orgzly can be triggered in multiple ways, so that should not be a problem.
That said, I'm reluctant to agree to any non-standard integration with any 3rd-party app that changes independently and can break Orgzly in the process. 
That said, is this issue still valid?

**Comment by Maltimore** [comment]
I feel like the main point of this issue (discussed in the couple of comments on the top) has been addressed with the introduction of automatically triggered sync at regular intervals. What was discussed a few comments further down should probably be tracked in separate issues, as it's very hard to say otherwise when this issue can be considered closed.
(I'm interested in improving the state of orgzly+syncthing too, that's why I'm interested in cleaning up the several issues that already track this)

**Comment by amberin** [comment]
Yes, I will now close this issue, as it is too broad and vague to ever finish. It's just too bad that so many people have voted for it...

To anyone coming here with Syncthing-related concerns: Please create a more specific issue (don't forget to search for existing ones first).
